scipy/sparse/compressed.py

```diff
@@ -834,8 +846,10 @@ def _insert_many(self, i, j, x):
         # update attributes
         self.indices = np.concatenate(indices_parts)
         self.data = np.concatenate(data_parts)
-        nnzs = np.asarray(np.ediff1d(self.indptr, to_begin=0), dtype=idx_dtype)
-        nnzs[1:][ui] += new_nnzs
+        nnzs = np.empty(self.indptr.shape, dtype=idx_dtype)
+        nnzs[0] = idx_dtype(0)
+        nnzs[1:] = np.diff(self.indptr).astype(idx_dtype)
+        nnzs[ui+1] += new_nnzs
```

Comment:
Is the switch from `[1:][ui]` to `[ui+1]` indexing here actually related to the migration to `np.diff` or a stylistic preference?

That's the only thing here that kind of had me thinking.

Reply:
It got me thinking too. The original code starts with zero padding the diff. Then slices away that zero such that the index `ui` becomes the correct index mask. I didn't want to change the logic so instead I've shifted the indices up by 1 . 

I have the impression that shifting the indices is less expensive since it avoids creating a temporary array in the memory.

Reply:
What about this:

```python
# maybe come up with a better name than tmp
tmp = np.diff(self.indptr)
tmp[ui] += new_nnzs
nnzs[1:] = tmp
```

Reply:
Indeed, that's what I had in mind but I didn't want to touch the logic. If you think that's OK, that is my preference too.